src/core/style/layer_style_change.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2023 MapLibre contributors
+
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "layer_parameter.hpp"
+#include "layer_style_change_p.hpp"
+#include "types.hpp"
+
+#include <QtCore/QMetaProperty>
+#include <QtCore/QRegularExpression>
+
+#include <QMapLibre/Map>
+
+namespace {
+
+QList<QByteArray> getAllPropertyNamesList(const QObject *object) {
+    const QMetaObject *metaObject = object->metaObject();
+    QList<QByteArray> propertyNames(object->dynamicPropertyNames());
+    for (int i = metaObject->propertyOffset(); i < metaObject->propertyCount(); ++i) {
+        propertyNames.append(metaObject->property(i).name());
+    }
+    return propertyNames;
+}
+
+QByteArray formatPropertyName(const QString &name) {
+    static const QRegularExpression camelCaseRegex(QStringLiteral("([a-z0-9])([A-Z])"));
+    return QString(name).replace(camelCaseRegex, QStringLiteral("\\1-\\2")).toLower().toLatin1();
+}
+
+QByteArray formatPropertyName(const QByteArray &name) {
+    const QString nameAsString = QString::fromLatin1(name);
+    return formatPropertyName(nameAsString);
+}
+
+} // namespace
```

Comment:
Might be worth moving to a utils/helper file if we plan to re-use this logic for any other classes.

Reply:
Done.